runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,47 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler
+--- Uses `xdg-open`, `wslview`, and `open` depending on the operating system to
+--- attempt to open the URI
+---
+---
+---@param uri string Uri to be opened
+---   TODO: uri should accept vim.uri.Uri
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri)
+  local cmd
+
+  local ismac = vim.loop.os_uname().sysname == 'Darwin'
+  local iswin = vim.loop.os_uname().sysname:lower():find('windows')
```

Comment:
I wonder there just use `sysname:match('Windows')` not work for all windows version ?

Reply:
this is how it is done in other parts of the code base

Reply:
and you can just run `vim.loop.os_uname().sysname` this once right 
 ```lua
  local sysname = vim.loop.os_uname().sysname
  if sysname == 'Darwin' then
        open 
  elseif sysname:match('Windows') then
        foo
  else 
       bar 
  end
  ```